controllers/appgroup_controller.go

```diff
@@ -129,7 +133,12 @@ func (r *ApplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 				appGroup.Annotations[lastSuccessfulApplicationGroupKey] = ""
 			}
 			requeue = false
-			_ = r.cleanupWorkflow(ctx, logr, appGroup)
+			err = r.cleanupWorkflow(ctx, logr, appGroup)
+			if err != nil {
+				if err == ErrStartedReverseWorkflow {
+					requeue = true
+				}
+			}
```

Comment:
We should handle the other type of legit errors here

Reply:
Hmm. Previously we were ignoring errors returned from cleanupWorkflow, which is why I did not handle them, but let me know if this looks right.